src/share/cxx/cxx_f90_interface.cpp

```diff
@@ -170,6 +170,9 @@ void f90_push_forcing_to_cxx(F90Ptr elem_derived_FM, F90Ptr elem_derived_FT,
   sync_to_device(ft_f90, elements.m_ft);
 
   Tracers &tracers = Context::singleton().get_tracers();
+  if(tracers.fq.data() == nullptr) {
```

Comment:
I believe here you should allocate fq (and copy the fq data) only if the ftype requires it. As the code in this PR is currently written, I believe the first time any type of forcing is done, fq will get allocated. We want to allocate fq only if the forcing type requires it. fq is a an array whose size scales with QSIZE_D, which makes it the largest kind of array that gets allocated in Hommexx.

Reply:
I expected as much, I'll be making the change tomorrow